_posts/2020-08-07-g2p.md

```diff
@@ -148,11 +152,63 @@ mappings:
+
+### Rule Ordering
+
+The order of your rules in `g2p` really matters! This is because some rules can either create or remove the context for other rules to apply. In linguistics, these rule ordering patterns are usually talked about as either [feeding, bleeding, counter-feeding, or counter-bleeding](https://linguistics.stackexchange.com/questions/6084/whats-the-difference-between-counterbleeding-bleeding-and-feeding) relationships. There are potentially valid reasons to want to encode any of these types of relationships in your rules. 
+
+To illustrate a possible problem, let's consider a `g2p` mapping for language that converts 'a̱' to 'ə' and 'a' to 'æ'. 'a̱' is a sequence of a normal a followed by a combining macron below (\u0331). Because \u0331 ('a̱') is easily confusable with \u0332 ('a̲'), in order to follow the rule of thumb for [Unicode escape sequences](#unicode-escape-sequences), I'll write the rules as follows:
+
+| in | out |
+|---|---|
+| a | æ |
+| a\u0331 | ə |
+
+Now, assuming an input to this mapping of 'a̱' (a\u0331), we would get 'æ̱' (æ\u0331) instead of 'ə'. Why is that? Because the first rule applies and turns 'a' into 'æ' before the second rule has a chance to apply. This is called a 'bleeding' relationship. In order to avoid it, we would need to write our rules as follows:
+
+| in | out |
+|---|---|
+| a\u0331 | ə |
+| a | æ |
+
+With this ordering, our input of 'a̱' (a\u0331) would turn into 'ə' as we expect, and our input of 'a' would turn into æ also as expected. Try it out on the [G2P Studio](https://g2p-studio.herokuapp.com) if you don't believe me!
+
+### Unicode Escape Sequences
+
+Sometimes you need rules to convert from characters that either don't render very well, or render in a confusing way. In those cases, you can alternatively use [Unicode escape sequences](https://www.rapidtables.com/code/text/unicode-characters.html). For example, maybe you want to write a rule that converts the standard ASCII 'g' to the strict IPA Unicode /ɡ/. As you can likely see on your browser, these characters look very similar, but they're not! The ASCII 'g' is U+0067 and the strict IPA 'ɡ' is U+0261. So, you can write a rule as follows:
+
+
+| in | out |
+|---|---|
+| \u0067 | \u0261 |
```

Comment:
Maybe note the difference between Unicode "U+" notation and the (Python string?) `\uXXXX` notation?